### PR TITLE
Support Creating TFDataset from ImageSet/TextSet/FeatureSet

### PR DESCRIPTION
**Pull request by yangw1234**
This feature requires a few new features in ImageSet, TextSet, FeatureSet and BigDL's optimizer:

- [x] Support ImageSet read from a directory of images with Label #1269 
- [x] ImageSet and TextSet transform return a new ImageSet/TextSet #1280 
- [x] Support ImageSet with zero based Label #1287
- [x] Fix loss metric in Keras #1288

I'll submit other PRs to address these issues.

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala [hunk 1]
```diff
@@ -252,18 +254,15 @@ class TFValidationMethod(val valMethod: ValidationMethod[Float],
...
-
-      override def next(): Sample[Float] = {
-        val oldSample = prev.next()
-        val newSize = oldSample.getFeatureSize() ++ oldSample.getLabelSize()
-        Sample(oldSample.getData(), newSize, null)
-      }
-    }
+  override def transform(feature: ImageFeature): ImageFeature = {
+    val oldSample = feature[Sample[Float]](ImageFeature.sample)
+    val newSize = oldSample.getFeatureSize() ++ oldSample.getLabelSize()
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala]: Change label to feature?

**Comment by qiuxin2012** [comment]
Add comment for you public methods.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala (hunk 1 above)
Comment (yangw1234) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala]: Yes. features and labels are both feed into tensorflow graph at the same time to bypass bigdl's criterion mechanism and let tensorflow to do backprop using a single pass

**Comment by yangw1234** [comment]
added docs

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2194/
The failed case is because running out of disk. It passed in http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2193/

File: pyzoo/zoo/examples/tensorflow/tfpark/estimator_inception.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                                                      ImageRandomCrop(224, 224),
+                                                      ImageRandomPreprocessing(ImageHFlip(), 0.5),
+                                                      ImageChannelNormalize(
+                                                          0.485, 0.456, 0.406,
+                                                          0.229, 0.224, 0.225),
+                                                      ImageMatToTensor(to_RGB=True, format="NHWC"),
+                                                      ImageSetToSample(input_keys=["imageTensor"],
+                                                                       target_keys=["label"])
+                                                      ])
+            feature_set = FeatureSet.image_frame(image_set.to_image_frame())
```
Comment (jason-dai) [inline pyzoo/zoo/examples/tensorflow/tfpark/estimator_inception.py]: we should add an `image_set` method to `FeatureSet`

**Comment by hkvision** [comment]
Where is TextSet supported?

**Comment by yangw1234** [comment]
@hkvision https://github.com/intel-analytics/analytics-zoo/pull/1270/files#diff-6221d26ec561f2bd1828d125a7d5c81cR1036

**Comment by hkvision** [comment]
Oops. I see. But I didn't find any unit tests showing the usage.

**Comment by yangw1234** [comment]
@hkvision yes currently no unit test covering that feature. Honestly I did not know how to use TextSet, would you help me with that?
